doc/users/dflt_style_changes.rst

```diff
@@ -473,6 +473,36 @@ or by setting::
 in your :file:`matplotlibrc` file.
 
 
+``fill_between`` and ``fill_betweenx``
+--------------------------------------
+
+`~matplotlib.axes.Axes.fill_between` and
+`~matplotlib.axes.Axes.fill_betweenx` both follow the patch color
+cycle.
+
+.. plot::
+
+   import matplotlib.pyplot as plt
+   import numpy as np
+
+   fig, (ax1, ax2) = plt.subplots(1, 2, figsize=(6, 3))
+   th = np.linspace(0, 2*np.pi, 128)
+   N = 5
+
+   def demo(ax, extra_kwargs, title):
+       ax.set_title(title)
+       return [ax.fill_between(th, np.sin((j / N) * np.pi + th), alpha=.5, **extra_kwargs)
+               for j in range(N)]
+
+   demo(ax1, {}, '2.x')
+   demo(ax2, {'facecolor': 'C0'}, 'non-cycled')
+
+If the edge or face color is set, then the color is not cycled.
```

Comment:
This was an intentional choice, if the user provides _any_ color information do not cycle, but it will take very little to convince me otherwise.

This was driven in part by the way we have defaulted to `edgecolor='none'` so using the `color` kwarg to cycle would implicitly turn the edges back on which got me thinking about the different types of colors on the collections and hence landed here.